src/services/WalletService.ts

```diff
@@ -175,6 +176,7 @@ export default abstract class WalletService {
         return;
       }
     }
+    this.loadingBalances = false;
```

Comment:
I recall I suggested wrapping the whole method in a try and setting loadingBalances as false in the finally.

There are a couple other instances above where we should be setting it to false that are not properly covered.